src/components/markdown.tsx

```diff
@@ -138,11 +138,77 @@ const OPTIONS: MarkdownToJSX.Options = {
 };
 /* @TODO: Try to improve this typings here. */
 
+const parseTabs = (text: string) => {
+  const regex = /tab\[([^\]]+)\]\[([^\]]+)\]((?:.|[\r\n])*?)(?=(?:tab\[)|$)/g;
+  const lines = text.split(`\n`);
+  const tabs = [];
+  let lastLine = 0;
+  let match;
+
+  while ((match = regex.exec(text)) !== null) {
+    const label = match[1];
+    const range = match[2];
+
+    const [start, end] = range.split(`,`).map(Number);
+
+    const indexOfEndCodeMarkdown = lines.findIndex((line) => line === `\`\`\``);
+    const tabLines = lines.slice(0, indexOfEndCodeMarkdown);
+    const content = tabLines.slice(start, end);
+    tabs.push({
+      label,
+      content: `${
+        content.length > 0
+          ? `\`\`\`\n${content.join(`\n`)}\n\`\`\``
+          : `No Content`
+      }`,
+      range,
+    });
+
+    lastLine = indexOfEndCodeMarkdown + 1;
+    if (content.length === 0) return tabs;
+  }
+
+  if (lastLine < lines.length) {
+    const remainingLines = lines.slice(lastLine).join(`\n`).trim();
+    if (!tabs.length) return tabs;
+    if (remainingLines) {
+      tabs.push({
+        label: `rest`,
+        content: remainingLines,
+        range: null,
+      });
+    }
+  }
+
+  return tabs;
+};
 interface MarkdownProps {
   children: string;
 }
 
 const Markdown: React.FC<MarkdownProps> = ({ children }) => {
+  const [removedTabs, setRemovedTabs] = React.useState<string[]>([]);
```

Comment:
It will have a huge performance problem when more code snippets are added. I recommend to: 

1. Firstly check if the command added at all (some easy way or validate command by simple regex check).
2. Then, and only them start slicing snippet.
3. Use useMemo or memo to avoid doing it all the time.